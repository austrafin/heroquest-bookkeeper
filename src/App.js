import React, { useState } from "react";
import ArmoryPage from "./ArmoryPage";
import TabPanel from "./TabPanel";
import { AppBar, Tabs, Tab } from "@material-ui/core";
import { makeStyles } from "@material-ui/core/styles";
import GamePage from "./GamePage";

const a11yProps = (index) => {
  return {
    id: `simple-tab-${index}`,
    "aria-controls": `simple-tabpanel-${index}`,
  };
};

const useStyles = makeStyles((theme) => ({
  root: {
    flexGrow: 1,
    backgroundColor: "rgba(148, 133, 0, 0.603)",
  },
}));

const App = () => {
  const classes = useStyles();
  const [tabIndex, setTabIndex] = useState(0);

  return (
    <div className={classes.root}>
      <AppBar position="static">
        <Tabs
          value={tabIndex}
          onChange={(event, newIndex) => {
            setTabIndex(newIndex);
          }}
          aria-label="simple tabs example"
        >
          <Tab label="Game" {...a11yProps(0)} />
          <Tab label="Armory" {...a11yProps(1)} />
          <Tab label="Potions" {...a11yProps(2)} />
        </Tabs>
      </AppBar>
      <TabPanel value={tabIndex} index={0}>
        <GamePage />
      </TabPanel>
      <TabPanel value={tabIndex} index={1}>
        <ArmoryPage />
      </TabPanel>
      <TabPanel value={tabIndex} index={2}>
        Potions
      </TabPanel>
    </div>
  );
};

export default App;
