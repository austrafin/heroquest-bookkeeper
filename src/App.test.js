import React from "react";
import { shallow } from "enzyme";
import App from "./App";

describe("App component", () => {
  it("renders correctly", () => {
    const component = shallow(<App />);
    expect(component).toMatchSnapshot();
  });
});
