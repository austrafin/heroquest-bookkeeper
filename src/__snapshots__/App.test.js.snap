// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component renders correctly 1`] = `
<div
  className="makeStyles-root-1"
>
  <WithStyles(ForwardRef(AppBar))
    position="static"
  >
    <WithStyles(ForwardRef(Tabs))
      onChange={[Function]}
      value={0}
    >
      <WithStyles(ForwardRef(Tab))
        label="Game"
      />
      <WithStyles(ForwardRef(Tab))
        label="Armory"
      />
      <WithStyles(ForwardRef(Tab))
        label="Potions"
      />
    </WithStyles(ForwardRef(Tabs))>
  </WithStyles(ForwardRef(AppBar))>
  <TabPanel
    index={0}
    value={0}
  >
    <_default />
  </TabPanel>
  <TabPanel
    index={1}
    value={0}
  >
    <_default />
  </TabPanel>
  <TabPanel
    index={2}
    value={0}
  >
    Potions
  </TabPanel>
</div>
`;
